### ISSUE TITLE
Handling Overlapping Styles in `render` Function in field_note

### ISSUE THREAD

**Issue by anirudhaCodes** [charmbracelet/huh#111]
**Describe the bug**
The render function currently exhibits undesired behavior when handling overlapping styles, specifically with italic, bold, and code block styles.
The current behavior makes it challenging to render text when overlapping styles are present. This issue arises when closing tags for italic and bold styles coincide with the opening or closing tags of a code block. `This happens when any styles overlap with each other.`

## example:
**Code**
```
huh.NewNote().
	Title("").
	Description("*Jon, _Welcome to_ Joncafe*, How may we take your *order*?")
```
**Current Behaviour:**
<img width="380" alt="image" src="https://github.com/charmbracelet/huh/assets/155895067/5523f030-2056-4a0a-ab06-09cf7820b27b">

In above image, you can see that the "JonCafe" should be bold but it is not.

**expected behaviour:**
<img width="360" alt="image" src="https://github.com/charmbracelet/huh/assets/155895067/42c6d16e-9ad9-46f7-b979-b4a8a20e53b7">

**To Reproduce**
1. create a `NewNote`  field in a form with overlapping styles as shown in example above
2. Then run the form

**Desktop (please complete the following information):**
 - Device : Laptop
 - OS : WIN 11
 - go Version : go 1.21.6 windows/amd64

**Proposed solution**
Instead of resetting all the formatting using `\033[0m` , we can just the reset the formatting for that particular style. as shown in solution below:
```diff
func render(input string) string {
	var result strings.Builder
	var italic, bold, codeblock bool

	for _, char := range input {
		switch char {
		case '_':
			if !italic {
				result.WriteString("\033[3m")
				italic = true
			} else {
-				result.WriteString("\033[0m")
+				result.WriteString("\033[23m")
				italic = false
			}
		case '*':
			if !bold {
				result.WriteString("\033[1m")
				bold = true
			} else {
-				result.WriteString("\033[0m")
+				result.WriteString("\033[22m")
				bold = false
			}
		case '`':
			if !codeblock {
				result.WriteString("\033[0;37;40m")
				result.WriteString(" ")
				codeblock = true
			} else {
				result.WriteString(" ")
				result.WriteString("\033[0m")
				codeblock = false
				
+				if bold {
+					result.WriteString("\033[1m")
+				}
+				if italic {
+					result.WriteString("\033[3m")
+				}
			}
		default:
			result.WriteRune(char)
		}
	}

	// Reset any open formatting
	result.WriteString("\033[0m")

	return result.String()
}
```

Assign this to me.

**Comment by anirudhaCodes** [comment]
I guess this issue is solved by the PR no. 112

**Comment by maaslalani** [comment]
Thanks again @anirudhaCodes for the really great contribution. This was solved by #112.
